DecipherNow/gm-fabric-dashboard
# build: add git safety belts prior to publishing

Resolves #1349

## Proposed Changes

  - Add safety checks to our publish script to be sure that we have master checked out and that there aren't any local uncommitted changes
  
## How to Test

  - Run `npm run publish -- --dryRun` and confirm that the dry run works as before
  - Run `npm run publish` and confirm that it errors out for not having master checked out
  - Change my test to resolve the promise if we are not on master
  - Run `npm run publish` and confirm that it errors out for having local uncommitted changes
  - Locally commit your changes (being sure not to push them to origin) and run `npm test` to make sure you get to the first confirmation question. Hit ctrl+C to exit without taking any action
